constraint/pkg/client/client.go

```diff
@@ -358,13 +358,18 @@ func (c *Client) AddConstraint(ctx context.Context, constraint *unstructured.Uns
 		return resp, clienterrors.ErrNoDriver
 	}
 
-	changed, err := cached.AddConstraint(constraint)
+	constraintWDefaults, err := c.withDefaultParams(constraint, template)
+	if err != nil {
+		return resp, err
+	}
+
+	changed, err := cached.AddConstraint(constraintWDefaults)
```

Comment:
IIRC cached.AddConstraint also deepcopies... is there a way to coherently eliminate redundant deepcopies?

Reply:
It does! 

https://github.com/open-policy-agent/frameworks/blob/d82cbe189211c80b37f0d827ca813b39c9da05ce/constraint/pkg/client/template_client.go#L101-L110

the `cpy` above seems to actually be stored (if changed/ different).

the `cpy` made from the `withDefaultParams` should get garbage collected (eventually) when this func exists as it's not stored in its entirety in `driver.AddConstraint()` so I didn't think we'd take a performance hit. Is there a particular concern here?

---

That said, more than happy to make some changes. We could eliminate the deep cpy from template_client.AddConstraint?